ci/scripts/java_jni_manylinux_build.sh

```diff
@@ -84,6 +84,7 @@ cmake \
   -DARROW_PYTHON=${ARROW_PYTHON} \
   -DARROW_RPATH_ORIGIN=${ARROW_RPATH_ORIGIN} \
   -DARROW_SNAPPY_USE_SHARED=OFF \
+  -DARROW_S3=ON \
```

Comment:
Could you keep this list in alphabetical order?

Reply:
Could you add `: ${ARROW_S3:=ON}` and use it as `-DARROW_S3=${ARROW_S3}` like other variables such `ARROW_DATASET` and `ARROW_GANDIVA`.